### DISCUSSION TITLE
Feature request: Invalidate specific urls in the cache via requests

### DISCUSSION THREAD

**Discussion by jemayn** [umbraco/Umbraco.Cloud.Issues#816] [Ideas]
### Issue description

Hey there!

We have an enterprise site on Umbraco Cloud that we are currently using a separate CDN with to ensure uptime of the website if something goes wrong. We then use things like content published events to refresh the cache for specific pages when they are published - that way we can cache everything while also ensuring content is always up to date.

With the recent CDN introduced on Umbraco Cloud we've been looking at using that instead - however the client is not interested in waiting for cache invalidation, so until we have more granular control I don't see the Cloud CDN feature as a viable alternative.

Hope this can get added sometime in the future 🙂

**Comment by sitereactor** [comment]
Hi @jemayn thanks for the feature request. I have two follow up questions to get more clarity:
- Are you using Cache Everything, so its not only static assets, but also HTML that is cached?
- With regards to cache invalidation - is this related to publishing content and not having to wait for it to be available? (cache invalidation takes up to 30 seconds and can't be done faster - TTL can also be set fairly low, but that can of course have implications when updating the cache from the origin if the origin is not available).

**Comment by jemayn** [comment]
Hey @sitereactor 

Currently we are not using the Cloud CDN at all, but on our other setup we cache everything - also HTML. This allows us to also pull the Cloud site down for a bit for maintenance, etc - and the external cache remains alive (not forms, etc, but most of the site).

For cache invalidation, yes we use it when publishing - we also have custom event handlers whenever something is published that ensures all dependencies are published and has their cache busted - so if you fx replace an image in the media library we update that image, but also all pages the image is used on as the image will be cached.

I'm assuming setting a super low TTL would cause problems with refreshing the cache fx every 2 mins for an entire site? Would much rather cache it for hours and then bust the cache on specific pages when we need it.
